### PR TITLE
Add dlasv2 and tests

### PR DESCRIPTION
**Pull request by btracey**

### CODE REVIEW

File: native/dlasv2.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package native
+
+import "math"
+
+// Dlasv2 computes the singular value decomposition of a 2×2 matrix.
+//  [ csl snl] [F G] [csr -snr] = [ssmax     0]
```
Comment (kortschak) [inline native/dlasv2.go]: Lower case f, g and h.

File: native/dlasv2.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+// singular value. [cls, snl] and [csr, snr] are the left and right singular vectors.
+func (impl Implementation) Dlasv2(f, g, h float64) (ssmin, ssmax, snr, csr, snl, csl float64) {
+	ft := f
+	fa := math.Abs(ft)
+	ht := h
+	ha := math.Abs(h)
+	// pmax points to the largest element of the matrix in terms of absolute value.
+	// 1 if F, 2 if G, 3 if H.
+	pmax := 1
+	swap := (ha > fa)
```
Comment (kortschak) [inline native/dlasv2.go]: Drop parens.

File: native/dlasv2.go [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+			d := fa - ha
+			l := d / fa
+			if d == fa { // deal with inf
+				l = 1
+			}
+			m := gt / ft
+			t := 2 - l
+			mm := m * m
+			tt := t * t
+			s := math.Sqrt(tt + mm)
```
Comment (kortschak) [inline native/dlasv2.go]: Use `math.Hypot(t, m)` here?

File: native/dlasv2.go [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+			var r float64
+			if l == 0 {
+				r = math.Abs(m)
+			} else {
+				r = math.Sqrt(l*l + mm)
```
Comment (kortschak) [inline native/dlasv2.go]: Use `math.Hypot(l, m)` here?

File: native/dlasv2.go [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+			if mm == 0 {
+				if l == 0 {
+					t = math.Copysign(2, ft) * math.Copysign(1, gt)
+				} else {
+					t = gt/math.Copysign(d, ft) + m/t
+				}
+			} else {
+				t = (m/(s+t) + m/(r+l)) * (1 + a)
+			}
+			l = math.Sqrt(t*t + 4)
```
Comment (kortschak) [inline native/dlasv2.go]: If not using `math.Hypot`, use `tt` since you already have it.

**Comment by kortschak** [comment]
LGTM with minor comments.

File: native/dlasv2.go (hunk 1 above)
Comment (btracey) [inline native/dlasv2.go]: Done.

File: native/dlasv2.go (hunk 2 above)
Comment (btracey) [inline native/dlasv2.go]: Done.

File: native/dlasv2.go (hunk 3 above)
Comment (btracey) [inline native/dlasv2.go]: Done.

File: native/dlasv2.go (hunk 4 above)
Comment (btracey) [inline native/dlasv2.go]: Done. Thanks.

**Pull request merged**